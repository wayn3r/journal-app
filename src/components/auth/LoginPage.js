import { Link } from 'react-router-dom'
import { useDispatch, useSelector } from 'react-redux'
import { useForm } from 'hooks'
import { Form, Input } from '../forms'
import { GoogleLoginButton } from '../buttons/GoogleLoginButton'
import { startGoogleLogin, startLoginEmailPassword } from 'actions'
import { check } from 'helpers/check'
const validations = {
    email: ({ value, email }) => [
        check(!email.isEmpty(value), 'Email is required'),
        check(email.isEmail(value), 'Email is invalid'),
    ],
    password: ({ value, password }) => [check(!password.isEmpty(value), 'Password is required')],
}

export const LoginPage = () => {
    const dispatch = useDispatch()
    const { loading } = useSelector(state => state.auth)
    const [values, onChange, , validate] = useForm(
        {
            email: '',
            password: '',
        },
        validations
    )
    const { email, password } = values

    const handleLogin = e => {
        if (!validate(e)) return

        dispatch(startLoginEmailPassword(email, password))
    }
    const handleGoogleLogin = () => {
        dispatch(startGoogleLogin())
    }
    return (
        <div className='auth__container animate__animated animate__fadeIn'>
            <h1 className='auth__title'>Login</h1>
            <Form className='auth__form' onSubmit={handleLogin}>
                <Input
                    label='Email'
                    type='email'
                    value={email}
                    name='email'
                    invalid='Invalid email'
                    onChange={onChange}
                />
                <Input
                    type='password'
                    label='Password'
                    name='password'
                    value={password}
                    onChange={onChange}
                />
                <button className='button button--primary' disabled={loading} type='submit'>
                    {!loading ? (
                        'Login'
                    ) : (
                        <>
                            <i className='fa fa-spinner fa-spin'></i> Loading...
                        </>
                    )}
                </button>
            </Form>
            <div className='auth__social-networks'>
                <p>Login with social networks</p>
                <GoogleLoginButton onClick={handleGoogleLogin} />
            </div>

            <Link to='/auth/signup' className='link auth__link'>
                Create new account
            </Link>
        </div>
    )
}
