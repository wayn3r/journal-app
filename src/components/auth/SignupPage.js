import { Link } from 'react-router-dom'
import { useDispatch, useSelector } from 'react-redux'
import { startSignUp } from 'actions'
import { useForm } from 'hooks'
import { check } from 'helpers/check'
import { Form, Input } from '../forms'

const initial = {
    name: '',
    email: '',
    password: '',
    password2: '',
}
const validations = {
    name: ({ value, name }) => [
        check(!name.isEmpty(value), 'Name is required'),
        check(name.isLength(value, { min: 3 }), 'Name is too short'),
    ],
    email: ({ value, email }) => [
        check(!email.isEmpty(value), 'Email is required'),
        check(email.isEmail(value), 'Email is invalid'),
    ],
    password: ({ value, password }) => [
        check(!password.isEmpty(value), 'Password is required'),
        check(password.isLength(value, { min: 6 }), 'Password must be at least 6 characters'),
    ],
    password2: ({ value, form }) => [check(value === form.password, 'Passwords do not match')],
}

export const SignupPage = () => {
    const [values, onChange, , validate] = useForm(initial, validations)
    const { name, email, password, password2 } = values
    const dispatch = useDispatch()
    const { loading } = useSelector(state => state.auth)
    const handleSignin = e => {
        if (!validate(e)) return

        dispatch(startSignUp(name, email, password))
    }

    return (
        <div className='auth__container animate__animated animate__fadeIn'>
            <h1 className='auth__title'>Sign Up</h1>
            <Form className='auth__form' onSubmit={handleSignin}>
                <Input
                    label='Name'
                    type='text'
                    name='name'
                    value={name}
                    invalid='Invalid name'
                    onChange={onChange}
                />
                <Input
                    label='Email'
                    type='email'
                    name='email'
                    value={email}
                    invalid='Invalid email'
                    onChange={onChange}
                />
                <Input
                    label='Password'
                    type='password'
                    name='password'
                    value={password}
                    invalid='Invalid password'
                    onChange={onChange}
                />
                <Input
                    label='Confirm password'
                    type='password'
                    name='password2'
                    value={password2}
                    invalid='Invalid password'
                    onChange={onChange}
                />

                <button className='button button--primary' type='submit' disabled={loading}>
                    {!loading ? (
                        'Sign Up'
                    ) : (
                        <>
                            <i className='fa fa-spinner fa-spin'></i> Loading...
                        </>
                    )}
                </button>
                <Link to='/auth/login' className='link auth__link'>
                    Already registered?
                </Link>
            </Form>
        </div>
    )
}
