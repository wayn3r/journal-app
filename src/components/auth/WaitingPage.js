export const PleaseWaitPage = () => {
    return (
        <div className='please-wait'>
            <div className='please-wait__title'>Journal App</div>
            <div className='please-wait__message'>
                <i className='please-wait__icon'></i>
                <span className='please-wait__text'>Checking authentication, please wait...</span>
            </div>
        </div>
    )
}
