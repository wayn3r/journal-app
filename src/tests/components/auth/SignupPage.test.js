import { mount } from 'enzyme'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { MemoryRouter } from 'react-router-dom'

import { SignupPage } from 'components/auth/SignupPage'

// jest.mock('actions/auth.actions', () => ({
//     startGoogleLogin: jest.fn(),
//     startLoginEmailPassword: jest.fn(),
// }))

const middlewares = [thunk]
const mockStore = configureStore(middlewares)
const initialState = {
    auth: {},
    notes: {},
}
const store = mockStore(initialState)
store.dispatch = jest.fn()
const wrapper = mount(
    <Provider store={store}>
        <MemoryRouter>
            <SignupPage />
        </MemoryRouter>
    </Provider>
)

describe('Pruebas en <SignupPage />', () => {
    beforeEach(() => {
        jest.clearAllMocks()
    })
    test('debe renderizarse correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })
    test('debe hacer dispatch de la acciÃ³n respectiva', () => {
        const emailField = wrapper.find('input[name="email"]')
        emailField.simulate('change', { target: { value: '', name: 'email' } })

        wrapper.find('form').simulate('submit', { preventDefault() {} })

        expect(store.dispatch).not.toHaveBeenCalled()
    })
})
