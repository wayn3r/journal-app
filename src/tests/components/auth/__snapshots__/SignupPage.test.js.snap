// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SignupPage /> debe renderizarse correctamente 1`] = `
<div
  className="auth__container animate__animated animate__fadeIn"
>
  <h1
    className="auth__title"
  >
    Sign Up
  </h1>
  <form
    className="auth__form"
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="input-group"
    >
      <input
        autoComplete="none"
        className="input-group__input"
        name="name"
        onChange={[Function]}
        placeholder=" "
        type="text"
        value=""
      />
      <label
        className="input-group__label"
      >
        Name
      </label>
      <span
        className="input-group__invalid-text"
      >
        Invalid name
      </span>
    </div>
    <div
      className="input-group"
    >
      <input
        autoComplete="none"
        className="input-group__input"
        name="email"
        onChange={[Function]}
        placeholder=" "
        type="email"
        value=""
      />
      <label
        className="input-group__label"
      >
        Email
      </label>
      <span
        className="input-group__invalid-text"
      >
        Invalid email
      </span>
    </div>
    <div
      className="input-group"
    >
      <input
        autoComplete="none"
        className="input-group__input"
        name="password"
        onChange={[Function]}
        placeholder=" "
        type="password"
        value=""
      />
      <label
        className="input-group__label"
      >
        Password
      </label>
      <span
        className="input-group__invalid-text"
      >
        Invalid password
      </span>
    </div>
    <div
      className="input-group"
    >
      <input
        autoComplete="none"
        className="input-group__input"
        name="password2"
        onChange={[Function]}
        placeholder=" "
        type="password"
        value=""
      />
      <label
        className="input-group__label"
      >
        Confirm password
      </label>
      <span
        className="input-group__invalid-text"
      >
        Invalid password
      </span>
    </div>
    <button
      className="button button--primary"
      type="submit"
    >
      Sign Up
    </button>
    <a
      className="link auth__link"
      href="/auth/login"
      onClick={[Function]}
    >
      Already registered?
    </a>
  </form>
</div>
`;
